.github/workflows/python.yml

```diff
@@ -0,0 +1,113 @@
+name: Python Package
+
+on:
+  pull_request:
+  push:
+    branches:
+      - '*'
+    tags:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build YAMS-Lib and Python Wheel
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: 'recursive'
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: 6.0.x
+
+      - name: Install Python packages
+        run: python -m pip install --upgrade pip setuptools build
+
+      - name: Build YAMS-Lib into proper directory
+        run: dotnet publish YAMS-LIB -c Release -o yams_py/yams
+
+      - name: build
+        run: PYTHONWARNINGS=error python -m build
+
+      - name: Store the packages
+        uses: actions/upload-artifact@v2
+        with:
+          name: python-package-distributions
+          path: dist
+
+  test:
+    needs:
+      - build
+
+    runs-on: ubuntu-latest
+    name: Test Python ${{ matrix.python.version }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - {version: '3.10', wheel: 'cp310-cp310'}
+          - {version: '3.11', wheel: 'cp311-cp311'}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: remove code outside of wheel
+        run: rm -rf yams_py
+        shell: bash
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python.version }}
+
+      - name: Download all the dists
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
+      - name: Install Python packages
+        run: python -m pip install --upgrade pip
+
+      - name: install built wheel
+        run: python -m pip install "$(ls dist/*.whl)[test]"
+        shell: bash
+
+      - name: run pytest
+        run: python -m pytest yams_py_tests
```

Comment:
I would set where the tests are found in pytest's config: https://docs.pytest.org/en/7.1.x/reference/customize.html#pyproject-toml